fcrepo-clients/src/test/java/edu/unc/lib/dl/fcrepo4/WorkObjectIT.java

```diff
@@ -138,4 +140,21 @@ public void addPrimaryObjectAndSupplements() throws Exception {
 		BinaryObject suppFile = suppMember.getOriginalFile();
 		assertEquals(filenameS, suppFile.getFilename());
 	}
+	
+	@Test
+	public void addModsTest() throws Exception {
+		WorkObject obj = repository.createWorkObject(pid);
+		InputStream modsStream = new FileInputStream(new File("src/test/resources/vocabTest.xml"));
+		obj.addDescription(modsStream);
+		assertObjectExists(obj.getMODS().getPid());
+	}
+	@Test
+	public void addSourceMdTest() throws Exception {
+		WorkObject anotherObj = repository.createWorkObject(pid);
+		String sourceProfile = "some source md";
```

Comment:
should assert that the profile gets set and that the source MD object exists.  I'd also recommend checking that content makes it into the MODS and source md files, similar to how the addDataFileTest is checking the binary content.